### PR TITLE
[BUGFIX] endtime has to be indexed in UTC (3.1.x)

### PR DESCRIPTION
**Pull request by AndreasA**
this is especially important for endtime because otherwise
DocExpirationUpdateProcessorFactory might be too early
or too late in removing a document

### CODE REVIEW

File: Classes/FieldProcessor/Service.php [hunk 1]
```diff
@@ -66,20 +66,24 @@ public function processDocument(Apache_Solr_Document $document, array $processin
 				}
 
 				switch ($instruction) {
+					case 'timestampToUtcIsoDate':
+						$processor  = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('Tx_Solr_FieldProcessor_TimestampToUtcIsoDate');
+						$fieldValue = $processor->process($fieldValue);
+						break;
 					case 'timestampToIsoDate':
-						$processor  = t3lib_div::makeInstance('Tx_Solr_FieldProcessor_TimestampToIsoDate');
+						$processor  = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('Tx_Solr_FieldProcessor_TimestampToIsoDate');
```
Comment (irnnr) [inline Classes/FieldProcessor/Service.php]: Please avoid unrelated changes :)

File: Classes/Util.php [hunk 2]
```diff
@@ -120,23 +122,30 @@ public static function timestampToIso($timestamp) {
...
+		return $dateTime ? (int)$dateTime->format('U') : 0;
+	}
+
+	/**
+	 * Converts a date from unix timestamp to ISO 8601 format in UTC timezone.
+	 *
+	 * @param	integer	$timestamp unix timestamp
+	 * @return	string	the date in ISO 8601 format
+	 */
+	public static function timestampToUtcIso($timestamp) {
```
Comment (irnnr) [inline Classes/Util.php]: This method is missing in the 3.0.x PR

File: Classes/FieldProcessor/Service.php (hunk 1 above)
Comment (AndreasA) [inline Classes/FieldProcessor/Service.php]: OK, but I wanted to add the new one with the correct class name which is why I replaced the others but could have kept t3lib_div here

File: Classes/Util.php (hunk 2 above)
Comment (AndreasA) [inline Classes/Util.php]: https://github.com/TYPO3-Solr/ext-solr/pull/8/files#diff-99a1d1c23fd6b626eea8167b0dd97934R115

It is there?

Comment (irnnr) [inline Classes/Util.php]: ok sorry, must have missed it. my bad

File: Classes/FieldProcessor/Service.php (hunk 1 above)
Comment (irnnr) [inline Classes/FieldProcessor/Service.php]: don't worry about these kind of changes, will take care of that

**Comment by irnnr** [comment]
Thanks!
